tools/gyp/pylib/gyp/generator/make.py

```diff
@@ -1616,7 +1618,7 @@ def WriteTarget(self, spec, configs, deps, link_deps, bundle_deps,
       self.WriteDoCmd([self.output_binary], deps, 'touch', part_of_all,
                       postbuilds=postbuilds)
     else:
-      print "WARNING: no output for", self.type, target
+      print("WARNING: no output for", self.type, self.target)
```

Comment:
Why has target become self.target here?

Reply:
From upstream https://github.com/nodejs/node-gyp/pull/1334#issuecomment-429650411  __target__ was an _undefined name_ in this context because it was neither defined nor imported.  However __self.target__ is used above and a few lines below.  These _don't forget self_ issues are fairly common.

In a compiled language this would be easily caught as _compile-time_ error.  However in a dynamic language like Python, we either find them with a linter or they when they halt our scripts at runtime.

Reply:
Assumed as much but wanted to confirm it was intentional :)